File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/configurations/shell_script/InterpreterConfigurationSettings.java
package com.github.alexandrecarlton.idea.settings.layout.configurations.shell_script;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.nio.file.Path;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@JsonDeserialize(as = ImmutableInterpreterConfigurationSettings.class)
public interface InterpreterConfigurationSettings {

  Optional<Path> interpreterPath();

  Optional<String> interpreterOptions();

}

File: layout/src/main/kotlin/com/github/alexandrecarlton/idea/settings/dialog/configurations/shell_script/InterpreterConfigurationSettings.kt
package com.github.alexandrecarlton.idea.settings.dialog.configurations.shell_script

import com.fasterxml.jackson.annotation.JsonProperty
import java.io.File

data class InterpreterConfigurationSettings(

    @JsonProperty("Interpreter path")
    val interpreterPath: File? = null,

    @JsonProperty("Interpreter options")
    val interpreterOptions: String? = null
)
